### PR TITLE
bib: Added Filter Functionality

### PR DESCRIPTION
**Pull request by gianninapr**
Added a filter functionality that lets the user filter the bibliography displayed when using `bib.to_text()` function. An example would be setting the filter to 'Software' as such: 
`bib.to_text(filter = 'Software')` which would then filter all the available references tracked by bib that are labeled 'Software'.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,4 +3,5 @@ astropy-helpers
 matplotlib
 ads
 synphot
+ginga
```
[the remark this answers is not in the document]
Comment (gianninapr) [reply requirements.txt]: That was the intention, but we were getting build errors within this PR due to ginga. I added this line then as an attempt to resolve it. @mommermi what do you think?

[the remark this answers is not in the document]
Comment (mkelley) [reply requirements.txt]: Let me try to edit the ginga_plugins module instead: #93

[the remark this answers is not in the document]
Comment (gianninapr) [reply requirements.txt]: Thank you!

**Comment by mkelley** [comment]
OK, I merged #93 .

**Pull request merged**